src/compiler/checker.ts

```diff
@@ -5690,6 +5690,17 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     }
                 }
                 else {
+                    const isInstantiationExpressionType = !!(getObjectFlags(type) & ObjectFlags.InstantiationExpressionType);
+                    if (isInstantiationExpressionType) {
+                        const instantiationExpressionType = type as InstantiationExpressionType;
+                        if (isTypeQueryNode(instantiationExpressionType.node)) {
+                            return factory.cloneNode(instantiationExpressionType.node);

```

Comment:
Use `serializeExistingTypeNode` here. Specifically, we may need to do things like remap jsdoc syntax into TS syntax in the subtree (type arguments), fixup module specifiers in import types for the new context, or track if the symbol referenced by a query is still visible within the new context.

Reply:
(Specifically, you should probably do checks similar to what occur in `serializeTypeForDeclaration` to see if this type node is safe to reuse)

Reply:
Do you mean the following checks?

```ts
if (!isErrorType(type) && enclosingDeclaration) {
    const declWithExistingAnnotation = getDeclarationWithTypeAnnotation(symbol, enclosingDeclaration);
    if (declWithExistingAnnotation && !isFunctionLikeDeclaration(declWithExistingAnnotation) && !isGetAccessorDeclaration(declWithExistingAnnotation)) {
        // try to reuse the existing annotation
        const existing = getEffectiveTypeAnnotationNode(declWithExistingAnnotation) !;
        if (typeNodeIsEquivalentToType(existing, declWithExistingAnnotation, type) && existingTypeNodeIsNotReferenceOrIsReferenceWithCompatibleTypeArgumentCount(existing, type)) {
            const result = serializeExistingTypeNode(context, existing, includePrivateSymbol, bundled);
            if (result) {
                return result;
            }
        }
    }
}
```